components/feature/session/src/main/java/mozilla/components/feature/session/middleware/undo/UndoMiddleware.kt

```diff
@@ -83,10 +87,12 @@ class UndoMiddleware(
     ) {
         clearJob?.cancel()
 
-        val recoverableTabs = tabs.mapNotNull {
-            it as? TabSessionState
-        }.map {
-            it.toRecoverableTab()
+        val recoverableTabs = arrayListOf<RecoverableTab>()
+        tabs.forEach {
+            if (it is TabSessionState) {
+                val index = context.state.findTabIndex(it.id)
+                recoverableTabs.add(it.toRecoverableTab(index))
+            }
```

Comment:
nit: Same comment as above regarding `arrayList`. I think the previous code can just remain and we do the same call inside the `map()`?